### PR TITLE
Develop to production

### PR DESCRIPTION
**Pull request by allaway**
@kthrog - A lot is going on with this PR, but I think a review of the current staging instance would suffice. Can you do this? Thanks! 

This PR incorporates:

- updates that fix the feature to upload annotations to synapse
- new feature to allow the usage of the display label, rather than the schema label, as the annotation key uploaded to synapse
- updated loading pages
- also a license, I think that came from the upstream repo. :)

### CODE REVIEW

File: app.R
Comment (allaway) [inline app.R]:
```suggestion
                     #folder_list <- syn_store$getStorageDatasetsInProject(synStore_obj, project_synID)
```

File: app.R
Comment (allaway) [inline app.R]:
```suggestion
                     folder_df <- syn_tableQuery(sprintf("select name, id from %s where type = 'folder' and projectId = '%s'", config$main_fileview, project_synID))$asDataFrame()
                     
                     for (i in seq_along(folder_list)) {
```

File: app.R
Comment (allaway) [inline app.R]:
```suggestion
                       folders_namedList <- setNames(as.list(folder_df$id), folder_df$name)
```

File: app.R
Comment (allaway) [inline app.R]:
```suggestion

```

File: app.R
Comment (allaway) [inline app.R]:
```suggestion
```

File: app.R
Comment (allaway) [inline app.R]:
```suggestion
               h3("1. Go to", strong("Select your Dataset"), "tab - select your project; choose the folder where your data are stored; and, choose your metadata template based on the type of data you’ll be annotating (e.g., Genomics Assay, Imaging Assay, etc.)."),
```

File: app.R
Comment (allaway) [inline app.R]:
```suggestion
              h3("2. Go to", strong("Get Metadata Template"), "tab - click the button to generate a metadata template in Google Sheets; once generated (this may take a few moments), click the link that appears, fill out the sheet (columns in blue are required; columns in yellow are conditionally required), and then download the file as a CSV."),
```

File: app.R
Comment (allaway) [inline app.R]:
```suggestion
                           h3("3. Go to", strong("Submit and Validate Metadata"), "tab - upload your filled CSV and validate your metadata. If you receive errors: correct them in the CSV, re-upload, and revalidate until you receive no more errors. Once your CSV is error-free, please submit it to nf-osi@sagebionetworks.org. (Please note: the submit to Synapse button is not currently functional!)"),
              h3("3. Go to", strong("Submit and Validate Metadata"), "tab - upload your filled CSV and validate your metadata. If you receive errors correct them, reupload your CSV, and revalidate until you receive no more errors. When your metadata is valid, you will be able to see a 'Submit' button. Press it to submit your metadata.")
	      h2("About"),
      ),
	      h3("The NF Data Curator App allows you to annotate your data files — stored in Sage Bionetworks’ data storage platform, Synapse — using metadata templates with preset attributes (a.k.a, keys or column headers) and values. The app also provides validation, to ensure metadata templates are completed according to the Neurofibromatosis Open Science Initiative’s (NF-OSI) metadata dictionary. Ultimately, this process enables data discovery on the NF Data Portal, once data are released."),
	      h2("Note"),
	      h3("Please note that this app is still in beta, and may not always work as designed. If you receive a ‘disconnected from the server’ error — this could mean that you’re not logged into Synapse (you must be logged in to use this app), or that something went wrong with the app — we apologize if this happens! You may have to reload the app and try again. If you experience repeated issues, please contact us at nf-osi@sagebionetworks.org. In addition, the third tab, Submit and Validate Metadata, is not fully functional. You won’t be able to submit your completed template to Synapse using this app. For now, please email your completed manifest to nf-osi@sagebionetworks.org.")     
      	      ),
```

**Comment by kthrog** [comment]
One thing I noticed before I merge, @allaway:

- Alphabetic sorting of the project, folder, and template dropdowns on the `Select Your Dataset` tab is no longer happening.... 

Not a big deal, but thought I'd mention it. Let me know if you want me to try to figure out where that happened, proceed anyway, or if you have an idea of what that might be. 

I also want to try uploading a completed template before I push to prod, but will have to do that in a bit.

File: app.R
Comment (allaway) [inline app.R]:
```suggestion
      updateSelectizeInput(session, 'var', choices = sort(names(projects_namedList)))
```

**Comment by allaway** [comment]
> Alphabetic sorting of the project, folder, and template dropdowns on the Select Your Dataset tab is no longer happening....

Good catch! This should be fixed now. This sorting is not in the upstream repository so it's easy to inadvertently lose when we pull in the other changes... :)

**Comment by kthrog** [comment]
So I don't think this should hold up the merge, but I'm wondering, as the code stands now, @allaway, does the metadata have to be fully validated correctly per our schema to submit to Synapse? Because that's what I'm experiencing right now trying to upload [this manifest](https://docs.google.com/spreadsheets/d/1RDxjSctfL3KTvW_d5dyhKZdK7O6kGGkhkv3W0aqj7Z8/edit#gid=0) to [this study](https://www.synapse.org/#!Synapse:syn11374337/wiki/488810). 

In theory, when we feel the metadata dictionary is more stable, this is how I want this function to work -- but right now, while we're still iterating so much, it's maybe not as ideal.

**Comment by allaway** [comment]
Hi Kaitlin,

Yes - the metadata must validate to show the upload button. If this looks
good to you, let's merge, and then I can take a stab at an override button.
:)

**Review by kthrog — APPROVED** [review]
This looks good! 

Some override ideas, off the top of my head, @allaway:
- I don't know if it's possible to introduce permissions-based features into the app, but it'd be neat if whenever the NF-Sage team is signed in, we can override to submit with errors.
- Or, alternatively: give users a 'But I promise my metadata is valid!'-like button where they override, but when they do, the full list of errors still showing is sent somewhere, maybe to the logs, or ideally, to us via email. And the user gets a notification: "You chose to override metadata validation and the NF-Sage team has been notified. We may be in touch with questions. Thanks!"

**Comment by allaway** [comment]
Thanks! I like option 1 better for now. Option two might be a good longer
term option, but until the schema is more comprehensive and well tested
from our end I imagine that folks would overuse this option.

**Comment by allaway** [comment]
I just updated the production app with these commits.

**Pull request merged**
